internal/endpoints/gitlab.go

```diff
@@ -0,0 +1,206 @@
+package endpoints
+
+import (
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"time"
+
+	"fmt"
+
+	"github.com/redhatinsights/platform-changelog-go/internal/config"
+	"github.com/redhatinsights/platform-changelog-go/internal/db"
+	l "github.com/redhatinsights/platform-changelog-go/internal/logging"
+	"github.com/redhatinsights/platform-changelog-go/internal/metrics"
+	m "github.com/redhatinsights/platform-changelog-go/internal/models"
+	"github.com/xanzy/go-gitlab"
+)
+
+func GetURL(p *gitlab.PushEvent) string {
+	if p == nil || p.Repository == nil {
+		return ""
+	}
+	return p.Repository.Homepage
+}
+
+func GetRepo(p *gitlab.PushEvent) *gitlab.Repository {
+	if p == nil || p.Repository == nil {
+		return nil
+	}
+	return p.Repository
+}
+
+func GetID(p *struct {
```

Comment:
Since you are using the same struct a few times. you can actually add this to this file as a `type` and just refernce it that way. like `type RepoInfo struct {<all the stuff>}`. Then you can just use `p *RepoInfo` here